### ISSUE TITLE
Send file not working in a very specific case

### ISSUE THREAD

**Issue by Martinlck** [evantahler/actionhero#968]
# Send file not working when the file is not found and the special 'if-modified-since' header is present

Description of Issue
The server crashes due to call a function on an undefined value

- ActionHero Version: latest
- Node.js Version: latest
- Operating System ubuntu 14

# Steps to reproduce your error
- Make a new actionhero project with `./node_modules/.bin/actionhero generate`
- Request for a file that does not exist using the special header
- See stack trace

Hello @evantahler, This time I come to you with a legit issue I think haha, Thanks again for AH is very amazing and I'm very happy with it.

If you allow me, I can try to submit a PR to fix this issue. Let me know.
Oh and also now that I have your attention, is there a way to not allow any file serving? Setting path.public to empty array will do it? 
Thanks again!

**Comment by evantahler** [comment]
Yes, please send in a PR! 
This is a funny case, because if the file doesn't exist, the browser shouldn't send a `'if-modified-since` header at all.  You can certainly end up in this situation due to deployments I guess. 

To disable file-serving entirely, you will want to do a few things:
- yes, remove any entries in `api.config.general.paths.public`
- however, then if someone goes to `/` on your host, you will get a 404.  You'll probably want to adjust the base routes for your API then, check out `api.config.servers.web.urlPathForActions`,  `api.config.servers.web.urlPathForFiles` and `api.config.servers.web.rootEndpointType`

**Comment by Martinlck** [comment]
Sure, I'll make some time today to submit one, thanks a lot!!!

**Comment by Martinlck** [comment]
Ok, I've been knocking my head a little bit because I started writing a test to reproduce the issue and couldnt reproduce it, and I also found a test that already should have reproduced the problem, but it wasnt happening. 
Then I went to the code on web.js inside servers and found out that on the repository is different than the npm module and the issue is fixed. So no need to do a PR :( I was excited to contribute... Well it will be next time! 
Thanks again for AH! And for your patience.
